app/controllers/application_controller.rb

```diff
@@ -7,6 +7,9 @@ class ApplicationController < ActionController::Base
   include SessionsHelper
   include CartsHelper
 
+  rescue_from CanCan::AccessDenied, with: :access_denied
+  rescue_from ActiveRecord::RecordNotFound, with: :not_found
```

Comment:
tại sao phải cần line này, nguyên nhân j sẽ raise record not found

Reply:
Khi dùng load_resource thì nó sẽ tự động load trước các tham số của resource. Ví dụ: PostController thì nó sẽ load trước @post. Vì một nguyên nhân nào đó, ví dụ như trong database bị xóa đột ngột hay không tìm thấy cái đối tượng đó thì hàm load này nó sẽ lỗi và cần nơi để bắt errors ạ. Nên em sử dụng line này